core/src/com/serwylo/beatgame/levels/customLevels.kt

```diff
@@ -0,0 +1,121 @@
+package com.serwylo.beatgame.levels
+
+import com.badlogic.gdx.Gdx
+import com.badlogic.gdx.files.FileHandle
+import com.google.gson.GsonBuilder
+import com.google.gson.annotations.SerializedName
+import com.google.gson.annotations.Since
+import com.mpatric.mp3agic.Mp3File
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.features.json.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.util.cio.*
+import io.ktor.utils.io.*
+
+
+private const val TAG = "customLevels"
+
+private const val CUSTOM_LEVELS_JSON_VERSION = 1.0
+
+private val gson = GsonBuilder().setPrettyPrinting().setVersion(CUSTOM_LEVELS_JSON_VERSION).create()
+
+fun customWorldFile(): FileHandle = Gdx.files.local("custom-worlds.json")
```

Comment:
This should probably be external storage, so it persists between installs. This does mean MP3 files copied there will hang around after an uninstall, which is not great, but people can manually remove it. The other alternative is people who set up a large collection of custom worlds will lose them when uninstalling and reinstalling.

In fact, perhaps the achievements should also be stored like that (though it does make the discoverability easier, letting people unlock levels - but do we care?).

Reply:
On second thoughts - uninstalling the game should properly remove these items. It removes everything else about an app typically (saved games, progress, etc) so why not customisations such as these.